src/redux/reducer.js

```diff
@@ -59,15 +61,47 @@ function loadApi(state = initialState.get('loadApi'), action) {
 
 function REPL(state = initialState.get('REPL'), action) {
   switch (action.type) {
+    case actType.GET_PREV_REPL_CODE:
+      //if codeIndex hasn't reached the begining of history and
+      //the previous code isn't an empty string
+      if (state.get('codeIndex') < state.get('history').size &&
```

Comment:
I'm not sure I'm a fan of how `codeIndex` works here. I would assume when `codeIndex=0`, you would be pointing at the beginning of the code history. But it looks like you have it set up the opposite way. What's your motivation for doing it that way?

Alternatively, you could change the name from `codeIndex` to something else. Usually `index` refers to a position in a list where 0 is the first item in the list.